### ISSUE TITLE
bug: IonTabButton1 href doesn't go to respective tab

### ISSUE THREAD

**Issue by TylerBurnett** [ionic-team/ionic-framework#21885]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x** React

**Current behavior:**

Any page that inherits <RouteComponentProps> from either React router OR React Router DOM will automatically overwrite the href URL of IonTabButton1, upon navigation to such page.

**Expected behavior:**

For that not to happen.

**Steps to reproduce:**

After installing the respective packages, take note of the current routing that is set in the App.tsx. Now once you start the app you should notice that you are on the intial tester page which in face inherits <RouteComponentProps>. Now attempt to navigate to IonTabButton1.
[Reproduction.zip](https://github.com/ionic-team/ionic-framework/files/5032203/Reproduction.zip)

**Related code:**

```
insert short code snippets here
```

**Other information:**

**Ionic info:**

```
Ionic:

   Ionic CLI       : 6.11.0 (C:\Users\tyler\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework : @ionic/react 5.3.1

Capacitor:

   Capacitor CLI   : 2.4.0
   @capacitor/core : 2.4.0

Utility:

   cordova-res : 0.14.1
   native-run  : not installed

System:

   NodeJS : v14.4.0 (C:\Program Files\nodejs\node.exe)
   npm    : 6.14.7
   OS     : Windows 10

```

**Comment by TylerBurnett** [comment]
Rolling back to @ionic/react@5.1.0 and @ionic/react-router@5.1.0 fixed the issue for the time being.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This appears to be regression in Ionic React v5.3.0. cc @elylucas

**Comment by tx-tim** [comment]
experiencing something very similar

**Comment by jlkiri** [comment]
Experiencing the similar issue.
Rolling back to @ionic/react@5.1.0 and @ionic/react-router@5.1.0 helps.

**Comment by tx-tim** [comment]
any update on this? Version 5.1 has issues with`goBack` from `NavContext`.  However, I can use `history.goBack` from `react-router-dom` to navigate back.  The problem with this is I don't have control over the animations

**Comment by djfooks** [comment]
If I roll back to @ionic/react@5.1.0 and @ionic/react-router@5.1.0 I get crashes in random bits of react-dom... which versions do I need for all the other packages?
I have this (generated from the template)
`  "dependencies": {
    "@capacitor/core": "2.4.3",
    "@ionic/react": "5.1.0",
    "@ionic/react-router": "5.1.0",
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.4.0",
    "@testing-library/user-event": "^8.0.3",
    "@types/jest": "^24.0.25",
    "@types/node": "^12.12.24",
    "@types/react": "^16.9.17",
    "@types/react-dom": "^16.9.4",
    "@types/react-router": "^5.1.4",
    "@types/react-router-dom": "^5.1.3",
    "ionicons": "^5.0.0",
    "react": "^16.13.0",
    "react-dom": "^16.13.0",
    "react-router": "^5.1.2",
    "react-router-dom": "^5.1.2",
    "react-scripts": "3.4.0",
    "typescript": "3.8.3"
  },`

**Comment by tx-tim** [comment]
@liamdebeasi does your fix for #22519 also fix this issue, or are they completely separate?

**Comment by liamdebeasi** [comment]
> @liamdebeasi does your fix for #22519 also fix this issue, or are they completely separate?

That is a separate fix that only affects Ionic Vue.

**Comment by sean-perkins** [comment]
Hello 👋  update surrounding this issue:

After reviewing the reproduction app, I identified that the usage of nested routes is incorrect. Currently the reproduction example has a page called `/tester` that is embedded within the `IonTabs` router outlet, what it is not a tab item nor a child page of a tab.

The routing definition should be refactored to:
```tsx
const App: React.FC = () => (
  <IonApp>
    <IonReactRouter>
      <IonRouterOutlet>
        <Route path="/tester" component={Tester} />
        <Route path="/" render={() => <Redirect to="/tester" />} exact={true} />
        <IonTabs>
          <IonRouterOutlet>
            <Route path="/tab1" component={Tab1} />
            <Route path="/tab2" component={Tab2} />
            <Route path="/tab3" component={Tab3} />
          </IonRouterOutlet>
          <IonTabBar slot="bottom">
            <IonTabButton tab="tab1" href="/tab1">
              <IonIcon icon={triangle} />
              <IonLabel>Tab 1</IonLabel>
            </IonTabButton>
            <IonTabButton tab="tab2" href="/tab2">
              <IonIcon icon={ellipse} />
              <IonLabel>Tab 2</IonLabel>
            </IonTabButton>
            <IonTabButton tab="tab3" href="/tab3">
              <IonIcon icon={square} />
              <IonLabel>Tab 3</IonLabel>
            </IonTabButton>
          </IonTabBar>
        </IonTabs>
      </IonRouterOutlet>
    </IonReactRouter>
  </IonApp>
);
```

The reason why this weird behavior is happening, is the tab bar will default the first tab as active (if not matched from the current url). When navigating away from `/tester` to `/tab2`, the logic for updating the tab button is thinking you are leaving `/tab1` instead and updates its path to return you to this non-child page `/tester` if you were to select tab 1 again.

TLDR; everything is working correctly on our end. I am going to close out this issue. I would recommend updating your routing definition first. If you experience any other issues while doing so, please open a new issue and I will be happy to investigate.
